src/agents/tests/matrix.sh

```
@@ -0,0 +1,350 @@
+    if [ "$got" = "$want" ]; then
+        pass "$name (count=$got)"
+    else
+        fail "$name" "expected $want occurrences of '$needle', got $got"
+    fi
+}
+
+RUN_ID="m$(date +%s)"
+cleanup_sessions() {
+    pkill -f "tools agents login" 2>/dev/null || true
+    sleep 0.3
+}
```

Shellcheck (0.11.0)

[info] This function is never invoked. Check usage (or ignored if invoked indirectly).

(SC2329)